Vienna/Sources/Main window/BrowserPane.m

```diff
@@ -779,4 +518,179 @@ -(BOOL)hasCloseButton
+    }
+}
+
+/* didReceiveTitle
+ * Invoked when the page title arrives. We use this to set the tab title.
+ */
+-(void)webView:(WebView *)sender didReceiveTitle:(NSString *)title forFrame:(WebFrame *)frame
+{
+    if (frame == self.webPane.mainFrame)
+    {
+        //TODO: do not use reference from browswerpane to tab
+        self.tab.label = title;
+        self.viewTitle = title;
+    }
+}
+
+/* didReceiveIcon
+ * Invoked when we get the page icon.
+ */
+-(void)webView:(WebView *)sender didReceiveIcon:(NSImage *)image forFrame:(WebFrame *)frame
+{
+    if (frame == self.webPane.mainFrame)
+    {
+        image.size = NSMakeSize(14, 14);
+        [self setIcon:image];
+        //TODO: show icon in tab (maybe based on user setting)
+    }
+}
+
+#pragma mark - WebFrameLoad helper methods
+
+-(void)setLoadingURL:(NSURL *)url unreachable:(NSURL *)unreachableURL {
+    if (!isLoading)
+    {
+        [self willChangeValueForKey:@"isLoading"];
+        isLoading = YES;
+        [self didChangeValueForKey:@"isLoading"];
+    }
+
+    // Show or hide the lock icon depending on whether this is a secure
+    // web page.
+    if ([url.scheme isEqualToString:@"https"])
+    {
+        [addressField.cell setHasSecureImage:YES];
+        [lockIconImage setHidden:NO];
+    }
+    else
+    {
+        [addressField.cell setHasSecureImage:NO];
+        [lockIconImage setHidden:YES];
+    }
+
+    if (!unreachableURL) {
+        self.url = url;
+    } else {
+        self.url = unreachableURL;
+    }
+}
+
+- (void)showError:(NSError *)error forWebFrame:(WebFrame *)frame {
+    [self setError:error];
+
+    // Use a warning sign as favicon
+    iconImage.image = [NSImage imageNamed:@"folderError.tiff"];
+
+    // Load the localized verion of the error page
+    NSString * pathToErrorPage = [[NSBundle bundleForClass:[self class]] pathForResource:@"errorpage" ofType:@"html"];
+    if (pathToErrorPage != nil)
+    {
+        NSString *errorMessage = [NSString stringWithContentsOfFile:pathToErrorPage encoding:NSUTF8StringEncoding error:NULL];
+        errorMessage = [errorMessage stringByReplacingOccurrencesOfString: @"$ErrorInformation" withString: error.localizedDescription];
+        if (errorMessage != nil)
+        {
+            [frame loadAlternateHTMLString:errorMessage baseURL:[NSURL fileURLWithPath:pathToErrorPage isDirectory:NO] forUnreachableURL:frame.provisionalDataSource.request.URL];
+        }
+        NSString *unreachableURL = frame.provisionalDataSource.unreachableURL.absoluteString;
+        if (unreachableURL != nil)
+            self.url = frame.provisionalDataSource.unreachableURL;
+    }
+}
+
+- (void)detectedRSSLink:(BOOL)detected {
+    hasRSSlink = detected;
+    [self showRssPageButton:detected];
+}
+
+#pragma mark - WebUIDelegate helper methods
+
+- (void)hoveredOverURL:(NSURL *)url {
+    if (self.statusBar) {
+        self.statusBar.label = url.absoluteString;
+    }
+}
+
+- (void)setIcon:(NSImage *)icon {
```

Comment:
Is it really a WebUIDelegate helper method ?
It's only called once, and from this class itself.

Reply:
You are right, it seems not to be a delegate method, just slipped into the wrong section. I probably confused it with `receivedIcon` from the `WebFrameLoadDelegte`

Reply:
These methods are in the `WebFrameLoadHelperMethods` section, not in the `WebUIDelegate` section

Reply:
Probably I should inline it.